### ISSUE TITLE
Promise-based usage

### ISSUE THREAD

**Issue by himynameisdave** [Keyang/node-csvtojson#65]
Each time I've used this module, I've ended up wrapping it in a promise which I find is a lot cleaner and more ES6exy.

Here's an example:

``` javascript
// require fs module
const fs = require('fs'),

      // typical csvtojson stuff, as outlined in the readme file
      Converter = require('csvtojson').Converter,
      converter = new Converter({}),

     // new conversion function
     // takes a file
     // returns a promise
      convertToJson = function( file ){
        return new Promise( (res, rej) => {
          converter.on("end_parsed", (jsonData) => {
            if(!jsonData){
              rej("CSV to JSON conversion failed!")
            }
            res(jsonData);
          });
          fs.createReadStream(file).pipe(converter);
        });
      };

      // ...and you can use this promise like so:
      convertToJson( 'my-data.csv' )
        // handle the successful data conversion
        .then( data => {
          console.log(data);
        })
        // handle a rejected promise
        .catch( console.warn );
```

Although I'm new to promises, I still felt that people might like to see how this could be done. In addition, I was wondering if there are any plans to make the `Converter` part of the module just return a promise like that to make things a little easier to read. I know there might be some people using versions of node without promise support but in the future I think this might be neato :tada:  

Let me know if I can help or answer any questions, and also let me know if I did anything wrong/if anything could be improved in the way I'm doing things above. Cheers! :smile:

**Comment by Keyang** [comment]
Hi,
it is a good idea.
As Converter inherits from Stream so the api wont be very straightforward.
(new Converter()  returns a stream object rather than a promise object)
But it definitely one of the things to be added to roadmap.

~Keyang

**Comment by himynameisdave** [comment]
Right I'm still grasping the concept of streams. Cool stuff though thanks for the input!

**Comment by sibelius** [comment]
we can convert this stream in an async generator as well

**Comment by Keyang** [comment]
Implemented since version 2.0.0
